site/en/docs/extensions/migrating/remote-hosted-code/index.md

```diff
@@ -0,0 +1,88 @@
+
+In these cases, moving your remote hosted code to Manifest V3 requires no additional steps.
+
+## Use cases requiring migration {: use-cases }
+
+The following use cases previously relying on remote hosted code will require workarounds when migrating to Manifest V3:
+
+-   Using remote hosted third-party JavaScript (e.g. Google Analytics JS).
+-   Using a third-party library that relies on `eval` / `new Function(...)` (e.g. templating).
+-   Add / modify extension functionality without a new store submission.
+-   Updating filtering / blocking lists at runtime.
+
+In the rest of this guide we will discuss different migration strategies for these use cases.
+
+## Using remote hosted third-party JavaScript {: remote-hosted-js }
+
+In MV2 extensions it is possible to include remote hosted third-party JavaScript in content scripts, popups or background pages. For example, using Google Analytics in a pop up works by including the official Google tag via a `script` tag:
+
+This is no longer supported in Manifest V3 extensions and requires additional migration steps. There are different workarounds available for embedding third-party JavaScript into your extension:
+
+-   **Bundle third-party JS into your extension:** If you are using a popular framework like React or Bootstrap which you were previously loading from an external server, you can download the minified files, add them to your project and import them locally. In some cases, libraries, such as Firebase, will fetch additional code on a by need basis at runtime. In this case, you have to download all possible dynamic imports at build time. Here is an [example script that bundles Firebase using Rollup.js][7].
+-   **Embed third party JS in a sandboxed iframe:** some third-party libraries might rely on the use of `eval` or `new Function(...)` which is not supported in Manifest V3 extensions. These operations are still supported in sandboxed iframes. For example, if you need a third party library for rendering your popup HTML, you could run the external script inside a sandboxed iframe inside your popup. Inside the iframe you can still execute arbitrary JavaScript. For more details read the [guide on sandboxed iframes][3]. However, this approach does not work if the third party code requires access to your popup’s DOM.
+-   **Move functionality to your server:** communicating with remote servers is allowed. If functionality needs to change often, moving client-side functionally to your server can be a feasible workaround.
+-   **Look for other workarounds.** If the previous approaches don’t help with your use case you might have to either find an alternative solution (i.e. migrate to a different library) or find other ways to use the library's functionality.. For example, in the case of Google Analytics, you can switch to the Google measurement protocol instead of using the official remote hosted JavaScript version as described in our [Google Analytics 4 guide][8].
```

Comment:
```suggestion
-   **Look for other workarounds.** If the previous approaches don’t help with your use case you might have to either find an alternative solution (i.e. migrate to a different library) or find other ways to use the library's functionality. For example, in the case of Google Analytics, you can switch to the Google measurement protocol instead of using the official remote hosted JavaScript version as described in our [Google Analytics 4 guide][8].
```